.github/workflows/compile-queries.yml

```diff
@@ -0,0 +1,57 @@
+name: "Compile all queries using the latest stable CodeQL CLI"
+
+on:
+  push:
+    branches: [main] # makes sure the cache gets populated
+  pull_request:
+    branches:
+      - main
+      - "rc/*"
+
+jobs:
+  compile-queries:
+    runs-on: ubuntu-latest-xl
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      # calculate the merge-base with main, in a way that works both on PRs and pushes to main. 
+      - name: Calculate merge-base
+        id: merge-base
+        env:
+          BASE_BRANCH: ${{ github.base_ref }}
+        run: |
+          MERGE_BASE=$(git merge-base --fork-point origin/$BASE_BRANCH)
+          echo "merge-base=$MERGE_BASE" >> $GITHUB_ENV
+      - name: Cache CodeQL query compilation
+        uses: actions/cache@v3
+        with:
+          path: '*/ql/src/.cache'
+          # current GH HEAD first, merge-base second, generic third
+          key: codeql-stable-compile-${{ github.sha }}
+          restore-keys: |
+            codeql-stable-compile-${{ env.merge-base }}
+            codeql-stable-compile-
+      - name: install codeql
+        run: gh extension install github/gh-codeql
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: check formatting
+        run: gh codeql query format */ql/{src,lib,test}/**/*.{qll,ql} --check-only
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
It does need the token, because I'm using `gh`.  
The CodeQL invocation itself doesn't need the token, but the `gh` invocation around it does.  

I would need to add `codeql` directly to the `$PATH` to get around that. 
If I end up going that route then I should probably modify `.github/actions/fetch-codeql/action.yml` to fit my needs (make it configurable between nighty/release).

Reply:
Does `gh` require the token even if it does not need to access github.com ?

Reply:
Trying locally, it looks like `gh codeql` does not need a token for `codeql` commands. Only things like `list-versions` complain that I need to log in.

Reply:
That first `gh codeql` invocation installs the CodeQL CLI, so that definitely needs the token.  
The rest might not.  

I think I'll try to modify `.github/actions/fetch-codeql/action.yml`, then I'll definitely not need the token for those.

Reply:
Ah yes indeed, the first run of `gh codeql` downloads the CodeQL CLI. I normally run an explicit `gh codeql set-channel` command.